fw/http_limits.c

```diff
@@ -684,13 +684,6 @@ frang_http_domain_fronting_check(const TfwHttpReq *req, FrangAcc *ra)
 	if (!(TFW_CONN_TYPE(req->conn) & TFW_FSM_HTTPS))
 		return T_OK;
 
-	/*
-	 * Do not perform any SNI<=>authority validation if
-	 * tls_match_any_server_name is set.
-	 */
-	if (tfw_tls_allow_any_sni)
```

Comment:
Why do you delete this check? Previously when tls_match_any_server_name is true, we return T_OK. Why this behavior is changed?

Reply:
When `tfw_tls_allow_any_sni` is true, it means connection with empty sni or sni not listed in our vhost is "routed" to default vhost. In other words `tfw_tls_allow_any_sni` is the way to enable default vhost and it's not related to other vhosts. But when we have this check here and we want to check SNI vs authority for some vhosts and don't check for other we can't do this, because `tfw_tls_allow_any_sni` will override it.

Now I would prefer to introduce new knob only for domain fronting, that must be disabled by default. It will depends on `tfw_tls_allow_any_sni` also, but in the future we will remove `default` vhost and `tfw_tls_allow_any_sni` and will check SNI only when vhost is found by http tables.